soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -551,6 +555,19 @@ def ipc_command(data, ext_data):
         for i in range(0, 256):
             buf[i] = i
         hda_streams[stream_id].write(buf)
+    elif data == 12: # HDA PRINT
+        log.info("Doing HDA Print")
+        stream_id = ext_data & 0xFF
+        buf_len = ext_data >> 8 & 0xFFFF
+        hda_str = hda_streams[stream_id]
+        #hda_str.mem.seek(0)
+        pos = hda_str.mem.tell()
+        buf_data = hda_str.mem.read(buf_len).decode("utf-8", "replace")
+        log.info(f"DSP LOG MSG (idx: {pos}, len: {buf_len}): {buf_data}")
+        pos = hda_str.mem.tell()
+        if pos >= hda_str.buf_len*2:
```

Comment:
What is the logic here, why the check is for greater than twice buffer's length ?

Reply:
HDAStream (class) actually has 2 buffers of buf len right next to each other, these get flip/flopped as it reads things in. So this maybe is wrong, but it lets the strings be read in